### PR TITLE
Added wmake (OpenFOAM build system) support - revised

### PR DESCRIPTION
**Pull request by handrake0724**
As per rdnetto's comments, the code is updated.
Thanks.

### CODE REVIEW

File: config_gen.py [hunk 1]
```diff
@@ -303,6 +303,25 @@ def run(cmd, *args, **kwargs):
         print("\nRunning make...")
         run(make_args, env=env, **proc_opts)
 
+    elif(os.path.exists(os.path.join(project_dir, "Make/options"))):
+        print("Found OpenFOAM Make/options")
+        print("Preparing build directory...")
```
Comment (rdnetto) [inline config_gen.py]: The `print("Preparing build directory")` line isn't needed, as we're no longer creating a temporary directory.

File: config_gen.py [hunk 2]
```diff
@@ -303,6 +303,25 @@ def run(cmd, *args, **kwargs):
...
+
+        # Since icpc could not find directory in which g++ resides,
+        # set environmental variables to gcc to make fake_build operate normally.
+
+        env['WM_COMPILER']='Gcc'
+        env['WM_CC']='gcc'
+        env['WM_CXX']='g++'
+
+        # OpenFOAM source directory with Make/options
+        proc_opts['cwd'] = project_dir
```
Comment (rdnetto) [inline config_gen.py]: This is already set at line 170.

File: config_gen.py [hunk 3]
```diff
@@ -303,6 +303,25 @@ def run(cmd, *args, **kwargs):
         print("\nRunning make...")
         run(make_args, env=env, **proc_opts)
 
+    elif(os.path.exists(os.path.join(project_dir, "Make/options"))):
+        print("Found OpenFOAM Make/options")
+        print("Preparing build directory...")
+
+        make_args = ['wmake']
```
Comment (rdnetto) [inline config_gen.py]: Please use double quotes `"` for strings, for consistency.

**Comment by rdnetto** [comment]
(Note to self: resubmission of #58.)

One other thing: could you please add a link to an open-source project using OpenFOAM to the Test Cases section of the README?

**Comment by rdnetto** [comment]
Thanks for the PR, sorry it took me so long to merge this.

**Comment by peter-duanyu** [comment]
Hi handrake0724,
Could you please update it for OpenFOAM-4.x. Currently, it didn't work.

Peter

**Comment by peter-duanyu** [comment]
Hi handrake0724,
I finally find out how to solve this problem by adding some codes from your old version. 

elif(os.path.exists(os.path.join(project_dir, "Make/options"))):
        print("Found OpenFOAM Make/options")
        # OpenFOAM build system

        keys = os.environ.keys()
        WM = re.compile('^WM_')
        for key in keys:
            if(WM.match(key)):
                env[key] = os.environ[key]

        make_args = ["wmake"]

        # Since icpc could not find directory in which g++ resides,
        # set environmental variables to gcc to make fake_build operate normally.

        env['WM_COMPILER']='Gcc'
        env['WM_CC']='gcc'
        env['WM_CXX']='g++'

        print("\nRunning wmake...")
        run(make_args, env=env, **proc_opts)

**Comment by peter-duanyu** [comment]
Hi handrake0724,
I finally find a way to solve this problem by adding some codes from your old version. Here is what I write in config_gen.py and I run it with -c clang-4.0 flag which can solve the clang error. clang-4.0 is the version in my PC.

```
elif(os.path.exists(os.path.join(project_dir, "Make/options"))):
        print("Found OpenFOAM Make/options")
        # OpenFOAM build system
        keys = os.environ.keys()
        WM = re.compile('^WM_')
        for key in keys:
            if(WM.match(key)):
                env[key] = os.environ[key]

        make_args = ["wmake"]

        # Since icpc could not find directory in which g++ resides,
        # set environmental variables to gcc to make fake_build operate normally.

        env['WM_COMPILER']='Gcc'
        env['WM_CC']='gcc'
        env['WM_CXX']='g++'

        print("\nRunning wmake...")
        run(make_args, env=env, **proc_opts)
```

**Comment by handrake0724** [comment]
@peter-duanyu Sorry for the late reply. If you use -e option, it will work without applying your patch.

**Pull request merged**